### PR TITLE
[frontend/backend] Redesign of the atomic testings details screen (#642)

### PR DESCRIPTION
**Pull request by johanah29**

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/atomic_testing/AtomicTestingMapper.java [hunk 1]
```diff
@@ -31,7 +35,8 @@ private static AtomicTestingOutputBuilder getAtomicTestingOutputBuilder(Inject i
             .id(inject.getId())
             .title(inject.getTitle())
             .type(inject.getType())
-            .injectorContract(inject.getInjectorContract())
+            .tagIds(inject.getTags().stream().map(Tag::getId).toList())
+        .injectorContract(inject.getInjectorContract())
```
Comment (guillaumejparis) [inline openbas-api/src/main/java/io/openbas/atomic_testing/AtomicTestingMapper.java:39]: tab problem

File: openbas-api/src/main/java/io/openbas/atomic_testing/form/AtomicTestingInput.java [hunk 2]
```diff
@@ -3,14 +3,17 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import io.openbas.rest.inject.form.InjectDocumentInput;
+import lombok.AllArgsConstructor;
```
Comment (guillaumejparis) [inline openbas-api/src/main/java/io/openbas/atomic_testing/form/AtomicTestingInput.java:6]: used ?

File: openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingHeader.tsx [hunk 3]
```diff
@@ -40,13 +42,20 @@ const AtomicTestingHeader = () => {
   const { onLaunchAtomicTesting } = useContext(AtomicTestingResultContext);
 
   // Fetching data
-  const { atomic }: { atomic: AtomicTestingOutput } = useHelper((helper: AtomicTestingHelper) => ({
+  const { atomic, atomicDetail }: { atomic: AtomicTestingOutput, atomicDetail: AtomicTestingDetailOutput, } = useHelper((helper: AtomicTestingHelper) => ({
     atomic: helper.getAtomicTesting(atomicId),
+    atomicDetail: helper.getAtomicTestingDetail(atomicId!),
```
Comment (guillaumejparis) [inline openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingHeader.tsx:47]: atomicId!  
Are you sure that is useful ?

File: openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingHeader.tsx [hunk 4]
```diff
@@ -40,13 +42,20 @@ const AtomicTestingHeader = () => {
...
   }));
   useDataLoader(() => {
     dispatch(fetchAtomicTesting(atomicId));
+    dispatch(fetchAtomicTestingDetail(atomicId));
   });
 
+  const [tags, setTags] = useState(atomic.atomic_tags);
```
Comment (guillaumejparis) [inline openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingHeader.tsx:54]: why use a state when you already have tags in the store ?

File: openbas-front/src/admin/components/scenarios/scenario/HeaderTags.tsx [hunk 5]
```diff
@@ -9,22 +9,26 @@ import TagChip from '../../common/tags/TagChip';
...
   tags,
-  disabled,
+  disabled = false,
   updateTags,
 }) => {
   // Standard hooks
   const dispatch = useAppDispatch();
+  useDataLoader(() => {
+    dispatch(fetchTags());
+  });
```
Comment (guillaumejparis) [inline openbas-front/src/admin/components/scenarios/scenario/HeaderTags.tsx:30]: Why fetchTags but no get from the store ?

File: openbas-api/src/main/java/io/openbas/atomic_testing/AtomicTestingMapper.java [hunk 6]
```diff
@@ -82,6 +88,33 @@ public static AtomicTestingDetailOutput toDetailDto(Inject inject) {
 
   }
 
+  public static List<AtomicTestingExpectation> getAtomicTestingExpectations(ObjectNode content) {
+    List<AtomicTestingExpectation> atomicTestingExpectations = new ArrayList<>();
+    if (content.has("expectations")) {
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/atomic_testing/AtomicTestingMapper.java:93]: Why you don't use direclty ObjectMapper here ?

File: openbas-api/src/main/java/io/openbas/atomic_testing/AtomicTestingService.java [hunk 7]
```diff
@@ -162,6 +167,16 @@ public Inject createOrUpdate(AtomicTestingInput input, String injectId) {
     return injectRepository.save(injectToSave);
   }
 
+  public Inject updateAtomicTestingTags(
+      @PathVariable @NotBlank final String injectId,
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/atomic_testing/AtomicTestingService.java:171]: @PathVariable annotation is used only in API and not for service

File: openbas-api/src/main/java/io/openbas/atomic_testing/AtomicTestingService.java [hunk 8]
```diff
@@ -162,6 +167,16 @@ public Inject createOrUpdate(AtomicTestingInput input, String injectId) {
...
+      @Valid @RequestBody final AtomicTestingUpdateTagsInput input) {
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/atomic_testing/AtomicTestingService.java:172]: Same for @valid annotation

File: openbas-api/src/main/java/io/openbas/atomic_testing/form/AtomicTestingExpectation.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+import jakarta.persistence.Enumerated;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+
+@Getter
+@Setter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class AtomicTestingExpectation {
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/atomic_testing/form/AtomicTestingExpectation.java:16]: Why not using io.openbas.model.inject.form.Expectation direclty ?

File: openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingHeader.tsx [hunk 10]
```diff
@@ -65,8 +72,10 @@ const AtomicTestingHeader = () => {
         <Typography variant="h1" gutterBottom classes={{ root: classes.title }}>
           {atomic.atomic_title}
         </Typography>
+        {atomicDetail?.status_label
+          && <StatusChip status={atomicDetail?.status_label} />
```
Comment (guillaumejparis) [inline openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingHeader.tsx:76]: Like discussed pass status label in the main dto or not use a atomicdetail dto at all

File: openbas-front/vite.config.ts.timestamp-1714398982499-119da07db5259.mjs
Comment (guillaumejparis) [inline openbas-front/vite.config.ts.timestamp-1714398982499-119da07db5259.mjs:1]: to delete

**Review by guillaumejparis — APPROVED** [review]
